core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/impl/api/update.kt

```diff
@@ -40,10 +46,45 @@ internal fun <T, C> Update<T, C>.updateWithValuePerColumnImpl(selector: Selector
         }
     }
 
+/**
+ * Implementation for Update As Frame:
+ * Replaces selected column groups with the result of the expression only where the filter is true.
+ */
+internal fun <T, C, R> Update<T, DataRow<C>>.asFrameImpl(expression: DataFrameExpression<C, DataFrame<R>>): DataFrame<T> =
+    if (df.isEmpty()) df
+    else df.replace(columns).with {
+        val src = it.asColumnGroup()
```

Comment:
Could we rename it somehow related to the Column Group, it's not too easy to understand the algorithm in the last section?

Reply:
You mean like `asColumnGroup` instead of `asFrame`? That's more a design question. `ColumnGroup` is a specialized version of `DataFrame` (DF with name), so it would behave similarly. Let's discuss that later, for now I just fixed the behavior of the function already present.

Reply:
I mean the src variable